processing/src/test/java/io/druid/query/extraction/SubstringDimExtractionFnTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.query.extraction;
+import com.google.common.collect.Sets;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.druid.jackson.DefaultObjectMapper;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Set;
+
+/**
+ */
+public class SubstringDimExtractionFnTest
+{
+
+  private static final String[] testStrings = {
+      "apple",
+      "awesome",
+      "asylum",
+      "business",
+      "be",
+      "cool",
+      "a"
+  };
+
+  private static final String[] testPaths = {
+      "/druid/prod/historical",
+      "/druid/prod/broker",
+      "/druid/prod/coordinator",
+      "/druid/demo/historical",
+      "/druid/demo/broker",
+      "/druid/demo/coordinator",
+  };
+
```

Comment:
why are testStrings and testPaths , they seem to be used only in single methods.